sys/linux/i2c.txt

```diff
@@ -0,0 +1,44 @@
+include <uapi/linux/i2c-dev.h>
+include <uapi/linux/i2c.h>
+
+define I2C_MAX_BLOCK	(I2C_SMBUS_BLOCK_MAX + 2)
+
+resource fd_i2c[fd]
+
+syz_open_dev$I2C(dev ptr[in, string["/dev/i2c-#"]], id intptr, flags flags[open_flags]) fd_i2c
+
+ioctl$I2C_RETRIES(fd fd_i2c, cmd const[I2C_RETRIES], arg intptr)
+ioctl$I2C_TIMEOUT(fd fd_i2c, cmd const[I2C_TIMEOUT], arg intptr)
+ioctl$I2C_SLAVE(fd fd_i2c, cmd const[I2C_SLAVE], arg intptr)
+ioctl$I2C_SLAVE_FORCE(fd fd_i2c, cmd const[I2C_SLAVE_FORCE], arg intptr)
+ioctl$I2C_TENBIT(fd fd_i2c, cmd const[I2C_TENBIT], arg intptr)
+ioctl$I2C_PEC(fd fd_i2c, cmd const[I2C_PEC], arg intptr)
+
+ioctl$I2C_FUNCS(fd fd_i2c, cmd const[I2C_FUNCS], arg ptr[in, intptr])
+ioctl$I2C_RDWR(fd fd_i2c, cmd const[I2C_RDWR], arg ptr[in, i2c_rdwr_ioctl_data])
+ioctl$I2C_SMBUS(fd fd_i2c, cmd const[I2C_SMBUS], arg ptr[in, i2c_smbus_ioctl_data])
+
+i2c_smbus_ioctl_data {
+	read_write	int8
+	command		int8
+	size		int32
+	data		ptr[in, i2c_smbus_data]
+}
+
+i2c_smbus_data [
+	byte	int8
+	word	int16
+	block	array[int8, I2C_MAX_BLOCK]
+]
+
+i2c_rdwr_ioctl_data {
+	msgs	ptr[in, i2c_msg]
+	nmsgs	int32
```

Comment:
I think `msgs` should be a pointer to an array of `i2c_msg`s and `nmsgs` should be the length of that array.

Reply:
fixed